rend3-routine/shaders/src/opaque.wgsl

```diff
@@ -512,7 +515,33 @@ fn fs_main(vs_out: VertexOutput) -> @location(0) vec4<f32> {
             shadow_value = shadow_sample_pcf5(shadows, comparison_sampler, shadow_coords, shadow_ndc.z);
         }
 
-        color += surface_shading(light, pixel, v, shadow_value * pixel.ambient_occlusion);
+        // Calculate light source vector
+        let l = normalize(view_mat3 * -light.direction);
+
+        color += surface_shading(l, light.color, pixel, v, shadow_value * pixel.ambient_occlusion);
+    }
+
+    for (var i = 0; i < i32(point_lights.count); i += 1) {
+        let light = point_lights.data[i];
+
+        // Delta to light
+        let delta = (uniforms.view * light.position).xyz - vs_out.view_position.xyz;
+
+        // Distance
+        let d = length(delta);
+
+        // Attenuate from light and cusp at radius
+        // Derivative is 0 at both d = 0 and d = radius
+        // Source: https://lisyarus.github.io/blog/graphics/2022/07/30/point-light-attenuation.html
+        let s = saturate(d / light.radius);
+        let s2 = pow(s, 2.0);
```

Comment:
```suggestion
        let s2 = s * s;
```

The compiler should do this transform, but in case it can't, let's do it explicitly. `pow` is harder to deal with than a multiply